cornac/data/basket_dataset.py

```diff
@@ -0,0 +1,363 @@
+# Copyright 2023 The Cornac Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+
+from collections import OrderedDict, defaultdict
+
+import numpy as np
+
+from ..utils import get_rng
+from ..utils import validate_format
+from ..utils import estimate_batches
+
+
+class BasketDataset(object):
```

Comment:
Let's try to extend from Dataset object and only overwrite the necessary functions. It will help avoid a lot of redundancy and easy future maintenance.